core/src/main/resources/hudson/model/BuildTimelineWidget/build-timeline-widget.js

```diff
@@ -11,19 +11,27 @@ function getData(eventSource1, current, min, max) {
   }
   if (!eventSource1.loaded[current]) {
     eventSource1.loaded[current] = true;
-    new Ajax.Request("timeline/data/", {
-      method: "POST",
-      parameters: { min: current * interval, max: (current + 1) * interval },
-      onSuccess: function (t) {
-        if (t.status != 0) {
-          try {
-            eventSource1.loadJSON(JSON.parse(t.responseText), ".");
+    fetch(
+      "timeline/data/?" +
+        new URLSearchParams({
+          min: current * interval,
+          max: (current + 1) * interval,
+        }),
+      {
+        method: "POST",
+        headers: crumb.wrap({}),
+      }
+    ).then((t) => {
+      if (t.status !== 0) {
```

Comment:
What would happen if the response from the server was < 200 or >= 300? My reading of the original Prototype.js code is that since it used `onSuccess` (rather than `onComplete`), a server response that is < 200 or >= 300 would not be logged (because there is no `onFailure` handler), the `onSuccess` handler would not be invoked, and the `t.status != 0` check in the original code was redundant (because `onSuccess` wouldn't be invoked in the first place with a 0 status). Can you see a flaw in my reasoning?

Based on this reasoning, I think that the new code is not equivalent to the old code. To make it equivalent, we would need to do something like this:

```suggestion
      if (t.ok) {
```

That would make the new code equivalent to the old by doing nothing on a failing HTTP status code (including not logging anything).

(I would personally consider improving the code to at least log the failing HTTP status code in such a situation, but that is a pre-existing issue that I will not focus on in this review.)